crates/deltalake-core/src/kernel/snapshot/extract.rs

```diff
@@ -0,0 +1,177 @@
+use std::sync::Arc;
+
+use arrow_array::{
+    Array, ArrowNativeTypeOp, ArrowNumericType, BooleanArray, ListArray, MapArray, PrimitiveArray,
+    RecordBatch, StringArray, StructArray,
+};
+use arrow_schema::{ArrowError, DataType};
+
+use crate::{DeltaResult, DeltaTableError};
+
+pub(crate) trait ProvidesColumnByName {
```

Comment:
This file would benefit from a lot of Rustdoc, I can follow it right now because I have a lot of this state in my head, but I guarantee I'll forget why this all exists in 2 months :laughing:

Reply:
I guess this is true for most (all) added files. I will see that we have some comments for us.

Once larger goal is to only expose the `*Snapshot` to the outside world, and leave how we resolve all of this internal to the core crate.